### ISSUE TITLE
Pre-commit missing errors in staged files

### ISSUE THREAD

**Issue by Ryanf55** [adrienverge/yamllint#553]
Got it. I upgraded. You can use close this once the docs are updated. 

I did try running it, but it's not agreeing with running `yamllint` in CLI. Should I file another issue, or am I using it incorrectly? The files are staged. 

 New pre-commit config: 
 ```yaml
 ---
repos:
  - repo: https://github.com/adrienverge/yamllint.git
    rev: v1.29.0
    hooks:
      - id: yamllint
```     

Example file:
```yaml
a: b
c: d
```

```
$ yamllint .

./foo.yaml
  1:1       warning  missing document start "---"  (document-start)
  2:5       error    no new line character at the end of file  (new-line-at-end-of-file)
  
  $ pre-commit run
yamllint.................................................................Passed
```

_Originally posted by @Ryanf55 in https://github.com/adrienverge/yamllint/issues/551#issuecomment-1478487598_

**Comment by andrewimeson** [comment]
I just realized why this behaves this way. The yamllint errors are only warning level in this case.

```sh-session
$ cat << 'EOF' > test.yaml
---
a: b
c: d
EOF

$ git add test.yaml
$ pre-commit run --all-files
yamllint.................................................................Passed
```

but if we add an error-level problem to the file like a duplicate key, it works as you were hoping

```sh-session
$ cat << 'EOF' > test.yaml
---
a: b
a: b
c: d
EOF
$ git add test.yaml
$ pre-commit run --all-files
yamllint.................................................................Failed
- hook id: yamllint
- exit code: 1

test.yaml
  3:1       error    duplication of key "a" in mapping  (key-duplicates)
```

I think this actually counts as expected behavior. If you want yamllint to fail on warnings, you need to add `--strict / -s` to yamllint, so with pre-commit:

```yaml
---
repos:
  - repo: https://github.com/adrienverge/yamllint.git
    rev: v1.29.0
    hooks:
      - id: yamllint
        args: [--strict]
```

**Comment by Ryanf55** [comment]
Shouldn't the no newline error should still make pre-commit fail? It's an error, not a warning.

**Comment by andrewimeson** [comment]
Yes, it should. If you do a `git status` does it possibly show your file as still having changes to stage? If you have `git config apply.whitespace` set to `fix` and you staged with `git add -p`, it would fix the missing newline on the staged version of the file, but not the on-disk version.

I can't reproduce the newline issue if I drop the `--strict` and reset the file to have a missing newline.

```sh-session
$ printf 'a: b\nc: d' > test.yaml
$ git add test.yaml
$ pre-commit run --all-files
yamllint.................................................................Failed
- hook id: yamllint
- exit code: 1

test.yaml
  1:1       warning  missing document start "---"  (document-start)
  2:5       error    no new line character at the end of file  (new-line-at-end-of-file)
```

**Comment by Ryanf55** [comment]
That was it. Adding `strict` to the example config should solve this. Thanks!
